### PR TITLE
refine the node local collector

### PR DESCRIPTION
**Pull request by yan234280533**

### CODE REVIEW

File: pkg/ensurance/collector/cadvisor/cadvisor_linux.go [hunk 1]
```diff
@@ -25,69 +24,59 @@ import (
 	"github.com/gocrane/crane/pkg/utils"
 )
 
-const (
-	cadvisorCollectorName = "cadvisor"
-)
-
 func init() {
-	registerMetrics(cadvisorCollectorName, []types.MetricName{types.MetricNameContainerCpuTotalUsage, types.MetricNameContainerSchedRunQueueTime}, NewCadvisor)
+	//registerMetrics(cadvisorCollectorName, []types.MetricName{types.MetricNameContainerCpuTotalUsage, types.MetricNameContainerSchedRunQueueTime}, NewCadvisor)
```
Comment (mfanjie) [inline pkg/ensurance/collector/cadvisor/cadvisor_linux.go:28]: remove the init() function if it is not needed

File: pkg/ensurance/collector/cadvisor/cadvisor_linux.go [hunk 2]
```diff
@@ -25,69 +24,59 @@ import (
...
+func NewCadvisor(podLister corelisters.PodLister) types.Collector {
 
 	var includedMetrics = cadvisorcontainer.MetricSet{
 		cadvisorcontainer.CpuUsageMetrics:         struct{}{},
 		cadvisorcontainer.ProcessSchedulerMetrics: struct{}{},
 	}
 
-	var allowDynamic bool = true
-	var maxHousekeepingInterval time.Duration = 10 * time.Second
+	var allowDynamic = true
```
Comment (mfanjie) [inline pkg/ensurance/collector/cadvisor/cadvisor_linux.go:52]: short format `allowDynamic := true` is better when you define a variable with value.

File: pkg/ensurance/collector/collector.go [hunk 3]
```diff
@@ -103,22 +122,60 @@ func (s *StateCollector) UpdateCollectors() {
...
 			s.collectors.Delete(types.NodeLocalCollectorType)
 		}
+
+		// cadvisor need to stop manager before
+		if value, exists := s.collectors.Load(types.CadvisorCollectorType); !exists {
+			s.collectors.Delete(types.CadvisorCollectorType)
+			c := value.(types.Collector)
+			if err = c.Stop(); err != nil {
+				klog.Errorf("Failed to stop the cadvisor manager.")
+				//nothing to do
```
Comment (mfanjie) [inline pkg/ensurance/collector/collector.go:158]: comment is not needed if the code already shows the intention.

File: pkg/ensurance/collector/nodelocal/cpu.go [hunk 4]
```diff
@@ -1,10 +1,13 @@
 package nodelocal
 
 import (
+	"errors"
 	"fmt"
 	"math"
 	"time"
 
+	"github.com/shirou/gopsutil/load"
```
Comment (mfanjie) [inline pkg/ensurance/collector/nodelocal/cpu.go:9]: org import

File: pkg/ensurance/collector/nodelocal/cpu.go [hunk 5]
```diff
@@ -13,13 +16,13 @@ import (
...
-	cpuCollectorName = "cpu"
-	MAX_PERCENTAGE   = 100
-	MIN_PERCENTAGE   = 0
+	cpuCollectorName     = "cpu"
+	cpuLoadCollectorName = "cpuLoad"
 )
 
 func init() {
-	registerMetrics(cpuCollectorName, []types.MetricName{types.MetricNameCpuTotalUsage, types.MetricNameCpuTotalUtilization}, NewCPUCollector)
+	registerMetrics(cpuCollectorName, []types.MetricName{types.MetricNameCpuTotalUsage, types.MetricNameCpuTotalUtilization}, collectCPU)
```
Comment (mfanjie) [inline pkg/ensurance/collector/nodelocal/cpu.go:24]: registerCollector?
metrics generally means promethus collector

File: pkg/ensurance/collector/types/interface.go [hunk 6]
```diff
@@ -1,11 +1,11 @@
-package collector
+package types
```
Comment (mfanjie) [inline pkg/ensurance/collector/types/interface.go:1]: I prefer the previous way, interface should be in the upper folder of implementer, please check k8s code for reference.

File: pkg/ensurance/collector/types/types.go [hunk 7]
```diff
@@ -43,14 +42,24 @@ const (
 )
 
 const (
-	NodeLocalCollectorType CollectType = "node-local"
+	NodeLocalCollectorType     CollectType = "node-local"
```
Comment (mfanjie) [inline pkg/ensurance/collector/types/types.go:45]: types.go has special meaning in k8s, it define the api, maybe const.go is better than types.go?

File: pkg/ensurance/collector/cadvisor/cadvisor_linux.go (hunk 1 above)
Comment (yan234280533) [reply pkg/ensurance/collector/cadvisor/cadvisor_linux.go:28]: done~

File: pkg/ensurance/collector/nodelocal/cpu.go (hunk 4 above)
Comment (yan234280533) [reply pkg/ensurance/collector/nodelocal/cpu.go:9]: done~

File: pkg/ensurance/collector/types/types.go (hunk 7 above)
Comment (yan234280533) [reply pkg/ensurance/collector/types/types.go:45]: In k8s if it an type define is in types.go. if it only an const value, it in constants.go.

File: pkg/ensurance/collector/types/interface.go (hunk 6 above)
Comment (yan234280533) [reply pkg/ensurance/collector/types/interface.go:1]: done~

File: pkg/ensurance/collector/nodelocal/cpu.go (hunk 5 above)
Comment (yan234280533) [reply pkg/ensurance/collector/nodelocal/cpu.go:24]: I modified it.

**Pull request merged**